tests/integration/awslambda/test_lambda_integration.py

```
@@ -396,6 +440,308 @@ def test_python_lambda_subscribe_sns_topic(
+            role_arn = iam_create_role_with_policy(
+                RoleName=role,
+                PolicyName=policy_name,
+                RoleDefinition=lambda_role,
+                PolicyDefinition=s3_lambda_permission,
+            )
+
+            response = create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON37,
+                role=role_arn,
+                libs=TEST_LAMBDA_LIBS,
+            )
+
+            response["CreateFunctionResponse"]["FunctionArn"]
+
+            # test destination-config for create_event_source_mapping
+            kinesis_name = f"test-kinesis-{short_uid()}"
+            kinesis_client.create_stream(StreamName=kinesis_name, ShardCount=1)
+
+            result = kinesis_client.describe_stream(StreamName=kinesis_name)["StreamDescription"]
+            kinesis_arn = result["StreamARN"]
+
+            # wait for stream-status "ACTIVE"
+            status = result["StreamStatus"]
+            if status != "ACTIVE":
+
+                def check_stream_active():
+                    state = kinesis_client.describe_stream(StreamName=kinesis_name)[
+                        "StreamDescription"
+                    ]["StreamStatus"]
+                    if state != "ACTIVE":
+                        raise Exception(f"StreamStatus is {state}")
+
+                retry(check_stream_active, retries=6, sleep=1.0, sleep_before=2.0)
+
+            queue_event_source_mapping = sqs_create_queue()
+            queue_failure_event_source_mapping_arn = sqs_queue_arn(queue_event_source_mapping)
+            destination_config = {
+                "OnFailure": {"Destination": queue_failure_event_source_mapping_arn}
+            }
+
+            result = lambda_client.create_event_source_mapping(
+                FunctionName=function_name,
+                BatchSize=1,
+                StartingPosition="TRIM_HORIZON",
```

fwiw, we don't actually support `TRIM_HORIZON` yet 🙂 It just silently defaults back to `LATEST`. I'm working on a PR to add support for `TRIM_HORIZON`.

---

I'm also running the tests against AWS, and I had troubles with LATEST as it wouldn't pick up the data from the test.